src/components/dash-cards.js

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useState, useEffect, useContext } from 'react';
 import donate from '../assets/dashboard-donate-logo.svg';
 import request from '../assets/dashboard-request-logo.svg';
 import drop from '../assets/dashboard-lastdonate-logo.svg';
 import sandhour from '../assets/dashboard-daysleft-logo.svg';
 import heart from '../assets/dashboard-totaldonate-logo.svg';
 import { Link } from 'react-router-dom';
 import { withNamespaces } from 'react-i18next';
+import { firestore } from '../api/firebase.js';
+import { Context } from '../Context';
 function DashCards({ t }) {
+  const [state, dispatch] = useContext(Context);
+  let d = new Date();
+  const [donation, setDonation] = useState([]);
+  const DonationList = [];
+  const getDonation = async () => {
+    return firestore
+      .collection('Donation')
+      .where('userId', '==', state.setUser.uid)
+      .onSnapshot((snapshot) => {
+        snapshot.forEach((doc) => DonationList.push({ ...doc.data() }));
+        setDonation(DonationList);
+      });
+  };
+
+  useEffect(() => {
+    getDonation();
+  }, []);
```

Comment:
empty dependency list causes build to fail, pass getDonation after using useCallback